src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1110,231 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
+addQuadratureFunction(const std::string &field_name, mfem::QuadratureFunction *qf,
+                     const std::string &buffer_name,
+                     axom::sidre::IndexType offset)
+{
+   sidre::Group* grp = m_bp_grp->getGroup("qfields/" + field_name);
+   SLIC_ASSERT_MSG(grp != NULL, "qfield " << field_name << " does not exist");
+
+   const int numDofs = qf->Size();
+
+   if(qf->GetData() == nullptr) {
+      AllocNamedBuffer(buffer_name, offset + numDofs);
+   }
+   /*
+    *  Mesh blueprint for a scalar-based grid function is of the form
+    *    /qfields/field_name/qspace_order
+    *              -- scalar value of QuadratureFunction's QuadratureSpace::Order
+    *    /qfields/field_name/values
+    *              -- array of size numDofs
+    */
+
+   // Make sure we have the View "values".
+   sidre::View *vv = alloc_view(grp, "values");
+
+   // Describe and apply the "values" View.
+   // If the data store has buffer for field_name (e.g. AllocNamedBuffer was
+   // called, or it was loaded from file), use that buffer.
+   sidre::View *bv = named_buffers_grp()->getView(buffer_name);
+   if (bv)
+   {
+      SLIC_ASSERT_MSG(bv->hasBuffer() && bv->isDescribed(), "");
+
+      // named buffers always have offset 0
+      SLIC_ASSERT_MSG(bv->getSchema().dtype().offset() == 0, "");
+      SLIC_ASSERT_MSG(bv->getNumElements() >= offset + numDofs, "");
+
+      if (vv->isEmpty())
+      {
+         vv->attachBuffer(bv->getBuffer())
+         ->apply(sidre::DOUBLE_ID, numDofs, offset);
+      }
+
+      qf->NewDataAndSize(vv->getData(), numDofs);
+   }
+   else
+   {
+      // If we are not managing the grid function's data,
+      // create a view with the external data
+      vv->setExternalDataPtr(sidre::DOUBLE_ID, numDofs, qf->GetData());
+   }
+   SLIC_ASSERT_MSG((numDofs > 0 && vv->isApplied()) ||
+                   (numDofs == 0 && vv->isEmpty() && vv->isDescribed()),
+                   "invalid View state");
+   SLIC_ASSERT_MSG(numDofs == 0 || vv->getData() == qf->GetData(),
+                   "View data is different from QuadratureFunction data");
+   SLIC_ASSERT_MSG(vv->getNumElements() == numDofs,
+                   "View size is different from QuadratureFunction size");
+}
+
+// private method
+// Should only be called on mpi rank 0 ( or if serial problem ).
+void MFEMSidreDataCollection::
+RegisterQFieldInBPIndex(const std::string& field_name, mfem::QuadratureFunction *qf)
+{
+   sidre::Group *bp_field_grp = m_bp_grp->getGroup("qfields/" + field_name);
+   sidre::Group *bp_index_field_grp =
+      m_bp_index_grp->createGroup("qfields/" + field_name);
+
+   bp_index_field_grp->createViewString( "path", bp_field_grp->getPathName() );
+   bp_index_field_grp->copyView( bp_field_grp->getView("topology") );
+   bp_index_field_grp->copyView( bp_field_grp->getView("vdim") );
+   bp_index_field_grp->copyView( bp_field_grp->getView("qspace_order") );
+
+   // Fix me?: Should this be qf->GetVDim() instead?
```

Comment:
So, I do have at least one application where I would have vector-valued qfields (more for restart capabilities than anything else) rather than a case where you might have those vector fields saved off as individual components.

Reply:
Thanks @rcarson3 -- restart functionality is independent of the blueprint index.

From sidre/hdf5's perspective the data looks like an array of doubles, so everything should work as long as you can initialize an `mfem::QuadratureFunction` from this data.

Currently, the blueprint index is only used by VisIt (and possibly ascent ?). So I think the `vdim` is not currently used. Presumably, we'd want this to be used in the future, which is why I wanted feedback from @cyrush.

Reply:
I swapped it over to `qf->GetVDim()`, but we can revert it back if it's determined that's not the best way to go.